### PR TITLE
ENH: Port TestPluginBase to framework

### PR DESCRIPTION
**Pull request by jakereps**

### CODE REVIEW

File: qiime/plugin/testing.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        if plugin is not None:
+            self.plugin = plugin
+        else:
+            self.fail('%s is not a registered QIIME 2 plugin.' %
+                      self.package.split('.')[0])
+
+        # TODO use qiime temp dir when ported to framework, and when the
+        # configurable temp dir exists
+        self.temp_dir = tempfile.TemporaryDirectory(
+            prefix='q2-types-test-temp-')
```
Comment (thermokarst) [inline qiime/plugin/testing.py]: Can you add another class attr to specify this prefix?

File: qiime/plugin/testing.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        except KeyError:
+            self.fail(
+                "Could not find registered transformer from %r to %r." %
+                (from_type, to_type))
+
+        return transformer_record.transformer
+
+    def assertRegisteredSemanticType(self, semantic_type):
+        try:
+            semantic_type_record = self.plugin.types[semantic_type.name]
```
Comment (thermokarst) [inline qiime/plugin/testing.py]: :+1:

File: qiime/plugin/testing.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import qiime
+
+
+# TODO Split out into more specific subclasses if necessary.
+class TestPluginBase(unittest.TestCase):
+    package = None
+
+    def setUp(self):
+        plugin = None
+        package = None
```
Comment (thermokarst) [inline qiime/plugin/testing.py]: I'm not sure if this is necessary --- we don't need `package` if the `try/except` starting on 24 fails.

Comment (jakereps) [inline qiime/plugin/testing.py]: True, refactored a bit and forgot it there. 👍

**Comment by jakereps** [comment]
Thanks @thermokarst 👍 , everything should be addressed!

File: qiime/plugin/testing.py [hunk 4]
```diff
@@ -15,31 +15,30 @@
...
         for name, plugin_ in qiime.sdk.PluginManager().plugins.items():
             if plugin_.package == package:
                 plugin = plugin_
 
         if plugin is not None:
             self.plugin = plugin
         else:
-            self.fail('%s is not a registered QIIME 2 plugin.' %
-                      self.package.split('.')[0])
+            self.fail('%s is not a registered QIIME 2 plugin.' % package)
```
Comment (thermokarst) [inline qiime/plugin/testing.py]: 👍

**Comment by thermokarst** [comment]
Thanks @jakereps!

Now it would be great to update any plugins using the old harness:
- [`q2-types`](https://github.com/qiime2/q2-types/blob/master/q2_types/testing.py#L16)
- [`q2-phylogeny`](https://github.com/qiime2/q2-phylogeny/blob/master/q2_phylogeny/tests/test_fasttree.py#L12)
- [`q2-alignment`](https://github.com/qiime2/q2-alignment/blob/master/q2_alignment/test/test_mafft.py#L12)
- [`q2-demux`](https://github.com/qiime2/q2-demux/blob/master/q2_demux/test/test_transformer.py#L7)
- [`q2-feature-classifier`](https://github.com/qiime2/q2-types/blob/master/q2_types/testing.py#L16)

Would you mind opening up issues in those projects to indicate the need to switch over? Feel free to whip up some PRs too. Thanks!!

**Comment by jairideout** [comment]
Thanks @jakereps for taking care of all this! 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Pull request merged**